security_monkey/export/__init__.py

```diff
@@ -0,0 +1,160 @@
+                val = getattr(val, at)
+            values.append(str(val))
+
+        out += ",".join(values) + "\n"
+    return Response(out, mimetype='text/csv')
+
+
+@export_blueprint.route("/export/issues")
+@rbac.allow(roles=["View"], methods=["GET"])
+def export_issues():
+    args = {}
+    args['regions'] = request.args.get('regions', None)
+    args['accounts'] = request.args.get('accounts', None)
+    args['active'] = request.args.get('active', None)
+    args['names'] = request.args.get('names', None)
+    args['technologies'] = request.args.get('technologies', None)
+    args['searchconfig'] = request.args.get('searchconfig', None)
+    args['ids'] = request.args.get('ids', None)
+
+    for k, v in args.items():
+        if not v:
+            del args[k]
+
+    query = ItemAudit.query.join("item")
+    if 'regions' in args:
+        regions = args['regions'].split(',')
+        query = query.filter(Item.region.in_(regions))
+    if 'accounts' in args:
+        accounts = args['accounts'].split(',')
+        query = query.join((Account, Account.id == Item.account_id))
+        query = query.filter(Account.name.in_(accounts))
+    if 'technologies' in args:
+        technologies = args['technologies'].split(',')
+        query = query.join((Technology, Technology.id == Item.tech_id))
+        query = query.filter(Technology.name.in_(technologies))
+    if 'names' in args:
+        names = args['names'].split(',')
+        query = query.filter(Item.name.in_(names))
+    if 'active' in args:
+        active = args['active'].lower() == "true"
+        query = query.join((ItemRevision, Item.latest_revision_id == ItemRevision.id))
+        query = query.filter(ItemRevision.active == active)
+    if 'searchconfig' in args:
+        search = args['searchconfig']
+        query = query.filter(
+            (ItemAudit.issue.ilike('%{}%'.format(search))) |
+            (ItemAudit.notes.ilike('%{}%'.format(search))) |
+            (ItemAudit.justification.ilike('%{}%'.format(search))) |
+            (Item.name.ilike('%{}%'.format(search)))
+        )
+    if 'enabledonly' in args:
+        query = query.join((AuditorSettings, AuditorSettings.id == ItemAudit.auditor_setting_id))
+        query = query.filter(AuditorSettings.disabled == False)
+
+    query = query.order_by(ItemAudit.justified, ItemAudit.score.desc())
+
+    attributes = [
+        ["item", "technology", "name"],
+        ["item", "account", "name"],
+        ["item", "account", "s3_name"],
+        ["item", "account", "number"],
+        ["item", "region"],
+        ["item", "name"],
+        ["item", "comments"],
+        ["score"],
+        ["issue"],
+        ["notes"],
+    ]
+
+    out = ",".join(["/".join(at) for at in attributes]) + "\n"
+
+    for issue in query:
+        values = []
+        for attribute in attributes:
+            val = issue
+            for at in attribute:
+                val = getattr(val, at)
+            values.append('"'+str(val)+'"')
```

Comment:
http://stackoverflow.com/questions/769621/dealing-with-commas-in-a-csv-file

The double quotes you're adding will help if the value has a comma, or a CRLF, but not if the value has a double quote:

https://tools.ietf.org/html/rfc4180

> (6.) Fields containing line breaks (CRLF), double quotes, and commas
>        should be enclosed in double-quotes.  For example:
> 
>        "aaa","b CRLF
>        bb","ccc" CRLF
>        zzz,yyy,xxx
> 
>
> (7.)  If double-quotes are used to enclose fields, then a double-quote
>       appearing inside a field must be escaped by preceding it with
>       another double quote.  For example:
>
>       "aaa","b""bb","ccc"

So I'm going to add a line after 156 replacing any `"` in val with `""`